subprojects/code-quality/src/integTest/groovy/org/gradle/api/plugins/quality/PmdPluginBuildCacheIntegrationTest.groovy

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.plugins.quality
+
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+import org.gradle.integtests.fixtures.DirectoryBuildCacheFixture
+import org.gradle.util.Requires
+import org.gradle.util.TestPrecondition
+
+import static org.hamcrest.Matchers.containsString
+import static org.hamcrest.Matchers.not
+
+class PmdPluginBuildCacheIntegrationTest extends AbstractIntegrationSpec implements DirectoryBuildCacheFixture {
+
+    def setup() {
+        buildFile << """
+            apply plugin: "java"
+            apply plugin: "pmd"
+            pmd {
+                incrementalAnalysis = true
+            }
+
+            ${mavenCentralRepository()}
+
+            ${!TestPrecondition.FIX_TO_WORK_ON_JAVA9.fulfilled ? "sourceCompatibility = 1.6" : ""}
+        """.stripIndent()
+    }
+
+    @Requires(TestPrecondition.NOT_WINDOWS)
+    def "pmd can work with build cache"() {
```

Comment:
❌ Note that you are not exercising the build cache at all during that test since you never run `clean`. You only test incremental build. Moreover, you need use the `withBuildCache` method to enable the build cache, e.g. `withBuildCache().succeeds "pmdMain"`.

Reply:
Thanks for reminding! Won't happen again.